### PR TITLE
[PWX-33959]: T34542904- Newly set pool labels should persist post pool expand using resize-disk

### PR DESCRIPTION
**Pull request by ragsingh-px**
**What this PR does / why we need it**:
Automation test to check if the pool labels persist before and after pool expand operation
**Which issue(s) this PR fixes** (optional)
Closes # PWX-33959

**Special notes for your reviewer**:
Testing link:- https://jenkins.pwx.dev.purestorage.com/user/ragsingh@purestorage.com/my-views/view/all/job/Users/job/Raghvendra/job/raghav_pxdv-7-dev/48/

### CODE REVIEW

File: tests/basic/pool_function_test.go [hunk 1]
```diff
@@ -675,3 +676,62 @@ var _ = Describe("{PoolExpandResizePoolMaintenanceCycle}", func() {
...
+	})
+
+	It("Initiate pool expansion and Newly set pool labels should persist post pool expand resize-disk operation", func() {
+		log.InfoD("set pool label, before pool expand")
+		labelBeforeExpand := poolToResize.Labels
+		poolLabelToUpdate := make(map[string]string)
+		poolLabelToUpdate["cust-type"] = "test-label"
+		// Update the pool label
+		err = Inst().V.UpdatePoolLabels(*storageNode, poolIDToResize, poolLabelToUpdate)
+		log.FailOnError(err, "Failed to update the label on the pool %s", poolIDToResize)
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:716]: Should this be dash.VerifyFatal()?

File: tests/basic/pool_function_test.go [hunk 2]
```diff
@@ -675,3 +676,62 @@ var _ = Describe("{PoolExpandResizePoolMaintenanceCycle}", func() {
...
+		log.InfoD("Current Size of the pool %s is %d GiB. Trying to expand to %v GiB with type resize-disk",
+			poolIDToResize, poolToResize.TotalSize/units.GiB, targetSizeGiB)
+		triggerPoolExpansion(poolIDToResize, targetSizeGiB, api.SdkStoragePool_RESIZE_TYPE_RESIZE_DISK)
+
+		err = waitForOngoingPoolExpansionToComplete(poolIDToResize)
+		dash.VerifyFatal(err, nil, "Pool expansion does not result in error")
+		verifyPoolSizeEqualOrLargerThanExpected(poolIDToResize, targetSizeGiB)
+
+		log.InfoD("check pool label, after pool expand")
+		labelAfterExpand := poolToResize.Labels
```
Comment (vinayakshnd) [inline tests/basic/pool_function_test.go:732]: `poolToResize` struct should be re-populated to fetch labels again. For this, call `poolToResize = getStoragePool(poolIDToResize)` again.

**Review by vinayakshnd — CHANGES_REQUESTED** [review]

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:732]: `poolToResize` is there in JustBeforeEach. I dont think its required fetch it inside.

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:716]: fixed.

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:990]: fixed.

**Review by vinayakshnd — APPROVED** [review]
LGTM

File: tests/basic/pool_function_test.go (hunk 1 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go]: fixed

File: tests/basic/pool_function_test.go (hunk 2 above)
Comment (ragsingh-px) [reply tests/basic/pool_function_test.go:1052]: fixed

**Pull request merged**